packages/kit/src/core/tsconfig.js

```diff
@@ -0,0 +1,101 @@
+
+	const paths = {};
+
+	paths['$lib'] = [path.relative(SVELTE_KIT, config.kit.files.lib)];
+	paths['$lib/*'] = [path.relative(SVELTE_KIT, config.kit.files.lib) + '/*'];
+
+	if (user_file) {
+		// we have to eval the file, since it's not parseable as JSON (contains comments)
+		const user_tsconfig_json = fs.readFileSync(user_file, 'utf-8');
+		const user_tsconfig = (0, eval)(`(${user_tsconfig_json})`);
+
+		// we need to check that the user's tsconfig extends the framework config
+		const extend = user_tsconfig.extends;
+		const extends_framework_config = extend && path.resolve('.', extend) === out;
+
+		if (extends_framework_config) {
+			const { paths: user_paths } = user_tsconfig.compilerOptions || {};
+
+			if (user_paths) {
+				const lib = user_paths['$lib'] || [];
+				const lib_ = user_paths['$lib/*'] || [];
+
+				const missing_lib_paths =
+					!lib.some(
+						(/** @type {string} */ relative) => path.resolve('.', relative) === config.kit.files.lib
+					) ||
+					!lib_.some(
+						(/** @type {string} */ relative) =>
+							path.resolve('.', relative) === config.kit.files.lib + '/*'
+					);
+
+				if (missing_lib_paths) {
+					console.warn(
+						`\u001B[1m\u001B[93mYour compilerOptions.paths in ${user_file} should include the following:\u001B[39m\u001B[22m`
+					);
+					const relative = path.relative('.', config.kit.files.lib);
+					console.warn(`{\n  "$lib":["${relative}"],\n  "$lib/*":["${relative}/*"]\n}`);
+				}
+			}
+		} else {
+			let relative = path.relative('.', out);
+			if (relative.startsWith(SVELTE_KIT)) relative = './' + relative;
+
+			console.warn(
+				`\u001B[1m\u001B[93mYour ${user_file} should include the following:\u001B[39m\u001B[22m`
+			);
+			console.warn(`"extends": "${relative}"`);
+		}
+	}
+
+	mkdirp(SVELTE_KIT);
+
+	fs.writeFileSync(
+		`${SVELTE_KIT}/tsconfig.json`,
+		JSON.stringify(
+			{
+				compilerOptions: {
+					moduleResolution: 'node',
+					module: 'es2020',
+					lib: ['es2020', 'DOM'],
+					target: 'es2020',
+					// svelte-preprocess cannot figure out whether you have a value or a type, so tell TypeScript
+					// to enforce using \`import type\` instead of \`import\` for Types.
+					importsNotUsedAsValues: 'error',
+					// TypeScript doesn't know about import usages in the template because it only sees the
+					// script of a Svelte file. Therefore preserve all value imports. Requires TS 4.5 or higher.
+					preserveValueImports: true,
+					isolatedModules: true,
+					resolveJsonModule: true,
+					// To have warnings/errors of the Svelte compiler at the correct position,
+					// enable source maps by default.
+					sourceMap: true,
+					esModuleInterop: true,
+					skipLibCheck: true,
+					forceConsistentCasingInFileNames: true,
+					baseUrl: path.relative(SVELTE_KIT, '.'),
+					allowJs: true,
+					checkJs: true,
+					paths
+				},
+				include: ['../**/*.d.ts', '../**/*.js', '../**/*.ts', '../**/*.svelte'],
```

Comment:
I recall there was a feature recently request to be able to customize the location/name of `.svelte-kit` directory, which would potentially mess up some of the stuff here, like these relative paths. This seems like the sort of thing that it would be easy to forget about if we do implement a feature where that can be customized. Do you think it's worth it to make this more flexible now?

Reply:
yeah, in some places this PR is doing it right, but i got lazy with `include`. definitely needs fixing. though actually we can probably get away with this, since `baseUrl` is set:

```js
include: ['**/*.d.ts', '**/*.js', '**/*.ts', '**/*.svelte']
```

Similarly, `paths` should come out as e.g. `src/lib` rather than `../src/lib`, it turns out